lib/web/desktop_playback.go

```diff
@@ -170,10 +170,23 @@ func (pp *playbackPlayer) streamSessionEvents(ctx context.Context, cancel contex
 			if !errors.Is(err, context.Canceled) {
 				pp.log.WithError(err).Errorf("streaming session %v", pp.sID)
 			}
+
 			return
 		case evt := <-eventsC:
 			if evt == nil {
 				pp.log.Debug("reached end of playback")
+
+				msg, err := utils.FastMarshal(struct {
+					Message string `json:"message"`
+				}{Message: "end"})
+				if err != nil {
+					pp.log.WithError(err).Errorf("failed to marshal end message into JSON: %v", msg)
+				} else {
+					if _, err := pp.ws.Write(msg); err != nil {
+						pp.log.WithError(err).Error("failed to write \"end\" message over websocket")
+					}
+				}
```

Comment:
Why not just use a static string instead?
```suggestion
				if _, err := pp.ws.Write([]byte(`{"message":"end"}`)); err != nil {
					pp.log.WithError(err).Error("failed to write \"end\" message over websocket")
				}
```